src/radiation/radiation.cpp

```diff
@@ -204,92 +451,285 @@ PetscErrorCode ablate::radiation::Radiation::ComputeRHSFunction(PetscReal time,
+
+    if (log) {
+        VecGetDM(cellGeomVec, &cellDM);
+        DMPlexGetGeometryFVM(cellDM, nullptr, nullptr, &minCellRadius) >> checkError;
+        VecGetArrayRead(cellGeomVec, &cellGeomArray) >> checkError;
+    }
 
-        /** Put the irradiation into the right hand side function*/
+    for (PetscInt c = cellRange.start; c < cellRange.end; ++c) {            //!< This will iterate only though local cells
+        const PetscInt iCell = cellRange.points ? cellRange.points[c] : c;  //!< Isolates the valid cells
+
+        // make sure we are not working on a ghost cell
+        PetscInt ghost = -1;
+        if (ghostLabel) {
+            DMLabelGetValue(ghostLabel, iCell, &ghost);
+        }
+        if (ghost >= 0) {
+            continue;
+        }
+
+        /** Gets the temperature from the cell index specified */
+        DMPlexPointLocalFieldRead(subDomain->GetDM(), iCell, temperatureField.id, auxArray, &temperature);
+
+        /** Put the irradiation into the right hand side function */
         PetscScalar* rhsValues;
         DMPlexPointLocalFieldRead(subDomain->GetDM(), iCell, eulerFieldInfo.id, rhsArray, &rhsValues);
         PetscReal losses = 4 * sbc * *temperature * *temperature * *temperature * *temperature;
-        rhsValues[ablate::finiteVolume::CompressibleFlowFields::RHOE] += -kappa * (losses - intensity);
-
-        /** Add to the number of cells that has been counted*/
-        ncells++;
+        rhsValues[ablate::finiteVolume::CompressibleFlowFields::RHOE] += -kappa * (losses - origin[iCell].intensity);
+        if (log) {
+            DMPlexPointLocalRead(cellDM, iCell, cellGeomArray, &cellGeom) >> checkError;  //!< Reads the cell location from the current cell
+            //                printf("Cell: %" PetscInt_FMT " Intensity: %f\n", iCell, origin[iCell].intensity);  //!< Wrap in a statement which only prints the cells in the middle of the domain.
+            printf("%f %f %f %f\n", cellGeom->centroid[0], cellGeom->centroid[1], cellGeom->centroid[2], origin[iCell].intensity);
+        }
     }
+
     /** Cleanup*/
     VecRestoreArrayRead(rhs, &rhsArray);
-    VecRestoreArray(loctemp, &temperatureArray);
-    subDomain->RestoreFieldLocalVector(temperatureField, &vis, &loctemp, &vdm);
     RestoreRange(cellRange);
 
+    if (log) {
+        EndEvent();
+        VecRestoreArrayRead(cellGeomVec, &cellGeomArray) >> checkError;
+    }
+
     PetscFunctionReturn(0);
 }
 
-PetscReal ablate::radiation::Radiation::FlameIntensity(double epsilon, double temperature) { /** Gets the flame intensity based on temperature and emissivity*/
+PetscReal ablate::radiation::Radiation::FlameIntensity(double epsilon, double temperature) { /** Gets the flame intensity based on temperature and emissivity (black body intensity) */
     const PetscReal sbc = 5.6696e-8;                                                         //!< Stefan-Boltzman Constant (J/K)
     const PetscReal pi = 3.1415926535897932384626433832795028841971693993;
```

Comment:
It looks like we have multiple definitions of Pi and some other constants, can we merge them into a single state class in utilities.